modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/planner/NumericTypesPrecisionsTest.java

```diff
@@ -0,0 +1,230 @@
+        assertEquals(STRING_PRECISION, typeSys.getMaxPrecision(SqlTypeName.BINARY));
+        assertEquals(STRING_PRECISION, typeSys.getMaxPrecision(SqlTypeName.VARBINARY));
+
+        assertEquals(3, typeSys.getMaxPrecision(SqlTypeName.TINYINT));
+        assertEquals(5, typeSys.getMaxPrecision(SqlTypeName.SMALLINT));
+        assertEquals(10, typeSys.getMaxPrecision(SqlTypeName.INTEGER));
+        assertEquals(19, typeSys.getMaxPrecision(SqlTypeName.BIGINT));
+        assertEquals(7, typeSys.getMaxPrecision(SqlTypeName.REAL));
+        assertEquals(15, typeSys.getMaxPrecision(SqlTypeName.FLOAT));
+        assertEquals(15, typeSys.getMaxPrecision(SqlTypeName.DOUBLE));
+        assertEquals(DECIMAL_PRECISION, typeSys.getMaxPrecision(SqlTypeName.DECIMAL));
+
+        assertEquals(0, typeSys.getMaxPrecision(SqlTypeName.DATE));
+        assertEquals(3, typeSys.getMaxPrecision(SqlTypeName.TIME));
+        assertEquals(3, typeSys.getMaxPrecision(SqlTypeName.TIME_WITH_LOCAL_TIME_ZONE));
+        assertEquals(3, typeSys.getMaxPrecision(SqlTypeName.TIMESTAMP));
+        assertEquals(3, typeSys.getMaxPrecision(SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE));
+    }
+
+    /** */
+    @Test
+    public void testDefaultPrecision() {
+        assertEquals(1, typeSys.getDefaultPrecision(SqlTypeName.CHAR));
+        assertEquals(RelDataType.PRECISION_NOT_SPECIFIED, typeSys.getDefaultPrecision(SqlTypeName.VARCHAR));
+        assertEquals(1, typeSys.getDefaultPrecision(SqlTypeName.BINARY));
+        assertEquals(RelDataType.PRECISION_NOT_SPECIFIED, typeSys.getDefaultPrecision(SqlTypeName.VARBINARY));
+
+        assertEquals(3, typeSys.getDefaultPrecision(SqlTypeName.TINYINT));
+        assertEquals(5, typeSys.getDefaultPrecision(SqlTypeName.SMALLINT));
+        assertEquals(10, typeSys.getDefaultPrecision(SqlTypeName.INTEGER));
+        assertEquals(19, typeSys.getDefaultPrecision(SqlTypeName.BIGINT));
+        assertEquals(7, typeSys.getDefaultPrecision(SqlTypeName.REAL));
+        assertEquals(15, typeSys.getDefaultPrecision(SqlTypeName.FLOAT));
+        assertEquals(15, typeSys.getDefaultPrecision(SqlTypeName.DOUBLE));
+        assertEquals(DECIMAL_PRECISION, typeSys.getDefaultPrecision(SqlTypeName.DECIMAL));
+
+        assertEquals(0, typeSys.getDefaultPrecision(SqlTypeName.DATE));
+        assertEquals(0, typeSys.getDefaultPrecision(SqlTypeName.TIME));
+        assertEquals(3, typeSys.getDefaultPrecision(SqlTypeName.TIME_WITH_LOCAL_TIME_ZONE));
+        assertEquals(3, typeSys.getDefaultPrecision(SqlTypeName.TIMESTAMP));
+        assertEquals(0, typeSys.getDefaultPrecision(SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE));
+    }
+
+
+    /** */
+    @Test
+    public void testMaxNumericPrecision() {
+        assertEquals(DECIMAL_PRECISION, typeSys.getMaxNumericPrecision());
+    }
+
+    /** */
+    @Test
+    public void testMaxNumericScale() {
+        assertEquals(DECIMAL_SCALE, typeSys.getMaxNumericScale());
+    }
```

Comment:
Why do we need these tests? Values for max/default prcision are hardcoded in calcite and looks like we just compare one hardcoded value with another one.

Reply:
We manually implement several values. Like:

```
 @Override public int getMaxNumericScale() {
        return Short.MAX_VALUE;
    }

    /** {@inheritDoc} */
    @Override public int getMaxNumericPrecision() {
        return Short.MAX_VALUE;
    }
```
or
```
 @Override public int getDefaultPrecision(SqlTypeName typeName) {
        // Timestamps internally stored as millis, precision more than 3 is redundant. At the same time,
        // default Calcite precision 0 causes truncation when converting to TIMESTAMP without specifying precision.
        if (typeName == SqlTypeName.TIMESTAMP || typeName == SqlTypeName.TIME_WITH_LOCAL_TIME_ZONE)
            return 3;

        return super.getDefaultPrecision(typeName);
    }
```

It looks usefult to be sure that nothing would change. Or we can remove the tests. WDYT?